### DISCUSSION TITLE
Can't merge and use resources from dict of other project

### DISCUSSION THREAD

**Discussion by StefanKoell** [AvaloniaUI/Avalonia#13558] [General]
As briefly discussed in the telegram chat, I've created a minimal repro. I'm not sure if I'm doing something wrong but the I can't use resources from .axaml files through merged dictionaries from other projects.

On the Mac, I need to build twice. The first time I get an error message 
```
System.Xaml.XamlException: 'Node MergeResourceInclude is unable to resolve "avares://ResourcesLib/Resources1.axaml" path. Line 31, position 39.' Line number '31' and line position '39'.
```
The second time it can I don't get the error message but the image doesn't show though.

On Windows I always get the build error but the app starts anyway but also doesn't show the image.

[ResourceSample.zip](https://github.com/AvaloniaUI/Avalonia/files/13309437/ResourceSample.zip)

**Comment by timunie** [comment]
Looks like `MergeInclude` is only available if the resource is in the same project. Workaround would be `ResourceInclude`.

**Reply by maxkatz6** [reply]
Yes. And it's not really a workaround, it's pretty much the design. MergeInclude is a more specific feature for bottle-neck optimizations when resources are in the same project, which is also intentionally undocumented.

**Reply by beemobile4net** [reply]
I've just stumbled upon this issue myself.  This material creates confusion:
https://docs.avaloniaui.net/docs/guides/styles-and-resources/resources

It uses `MergeResourceInclude` as well as `ResourceInclude` as if they were interchangeable without any explanation which should be used when. Oh, don't take this as a negative criticism. I love Avalonia!

**Comment by StefanKoell** [comment]
Hi Tim!

Here's an update. When I include these resources from another project using ResourceInclude like you suggested, I'm not able to look it up in code behind (check the .cs file).

Am I doing it wrong? Is there another way to access these assets?

Thanks!

[ResourceSampleV2.zip](https://github.com/AvaloniaUI/Avalonia/files/13309993/ResourceSampleV2.zip)

**Reply by timunie** [reply]
```cs
Image1.Source = Application.Current!.FindResource("action-repeat-small") as DrawingImage;
```

> **NOTE**
> FindResource will also scan merged dictionaries while directly accessing Resources will not do this. Looks like this is missing in the docs. Will add a hint there.

**Reply by StefanKoell** [reply]
Thanks, Tim! That indeed does the trick. I guess TryGetResource will also scan merged dicts - at least in my tests and when I check the source code. The question is, is both supported or is one method the preferred one in case the internal API changes in the future?

**Reply by timunie** [reply]
https://github.com/AvaloniaUI/avalonia-docs/pull/236/files

**Reply by StefanKoell** [reply]
Awesome addition to the docs, Tim! This clears up everything for me. Thanks for your help again. I really appreciate it!
